### ISSUE TITLE
CI: run all.sh in a pure tf-psa-crypto environment

### ISSUE THREAD

**Issue by mpg** [Mbed-TLS/TF-PSA-Crypto#116]
**Context:** After https://github.com/Mbed-TLS/TF-PSA-Crypto/issues/6 the way we run the CI in tf-psa-crypto is through an Mbed TLS environment: we check out mbedtls, update tf-psa-crypto inside it, then run `all.sh` from Mbed TLS (which includes the crypto components).

Ultimately we want the CI (or at least the PR job) for tf-psa-crypto to be autonomous, ie run `all.sh` from a tf-psa-crypto checkout without checking out mbedtls. Right now we can't do just this, as `all.sh` from tf-psa-crypto doesn't have sufficient coverage yet (we're in the process of migrating/duplicating all relevant components to fix that).

During this migration period, we want to ensure that all the migrated component can actually run in a pure tf-psa-crypto environment (that is, without a surrounding mbedtls checkout). Without an automated check, it's all too easy for us to add a component that accidentally depends on files that are only present in mbedtls.

**This task** is to add a run of `all.sh` from tf-psa-crypto without a surrounding mbedtls checkout to tf-psa-crypto's PR CI. (While keeping the run of `all.sh` from mbedtls in addition for now.) Depending on what's more convenient, this may be either a separate job or the same job as the existing one.

**Depends on:** https://github.com/Mbed-TLS/mbedtls-framework/pull/80 and its companion PRs being merged (otherwise `all.sh` will fail pretty early in a pure tf-psa-crypto environment).

**Comment by mpg** [comment]
Note: once this is done, we can remove the temporary wrapper from mbedtls (that is, move `mbedtls-all.sh` back to `all.sh`). The purpose of that wrapper was to make mbedtls's all.sh list tf-psa-crypto components so that they would be executed (in an mbedtls environment). Once these components are executed in a pure tf-psa-crypto environment, I see no point executing them in an mbedtls environment (or as part of the mbedtls CI) too.

@ronald-cron-arm Not sure if you want to create a follow-up task for that, or just handle it as part of this task.

**Comment by ronald-cron-arm** [comment]
> Note: once this is done, we can remove the temporary wrapper from mbedtls (that is, move `mbedtls-all.sh` back to `all.sh`). The purpose of that wrapper was to make mbedtls's all.sh list tf-psa-crypto components so that they would be executed (in an mbedtls environment). Once these components are executed in a pure tf-psa-crypto environment, I see no point executing them in an mbedtls environment (or as part of the mbedtls CI) too.
> 
> @ronald-cron-arm Not sure if you want to create a follow-up task for that, or just handle it as part of this task.

That seems a small change thus I'd say just handle it as part of this task.
